src/mongo_client.ts

```diff
@@ -364,6 +369,32 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
         return true;
       }
     };
+    this.checkForNonGenuineHosts();
+  }
+
+  /** @internal */
+  private checkForNonGenuineHosts() {
+    if (this.mongoLogger && this.mongoLogger.logDestination) {
```

Comment:
The mongoLogger is always defined on the client (see src/mongo_client.ts:342). You also shouldn't need to check that the logDestination exists. It will always be set in the MongoLogger constructor (see src/mongo_logger.ts:479).

The severity logging methods check if a message is to be emitted on the component provided at the specified severity level, so we don't have to do these checks.

Reply:
ahh, I see, I'll remove these.